daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -140,7 +140,7 @@ class VariableInstance private (val rd: VariableRuntimeData)
 
 object VariableUtils {
 
-  def setExternalVariables(currentVMap: VariableMap, bindings: Seq[Binding], referringContext: ThrowsSDE) = {
+  def withExternalVariables(currentVMap: VariableMap, bindings: Seq[Binding], referringContext: ThrowsSDE) = {
     bindings.foreach { b => currentVMap.setExtVariable(b.varQName, b.varValue, referringContext) }
```

Comment:
This calls `setExternalVariable` which looks like it mutates the variable map. Because of that this should probably stay as `setExternalVariables` to make it clear this mutates state.

This also related to another comment I made about what calls this function, that comment probably no longer applies.

Note I wonder if there is a bug...do our API `withExternalVariables` functions in the DataProcessor end up calling this `setExternalVariables` function and mutate state instead of copying? We may need to investigate that, but that can be done separately.

Reply:
I checked daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala lines 297, 308, and 332 in the main branch.  They all have similar bodies:

```scala
     val newBindings = loadExternalVariables(extVars)
     copy(externalVars = newBindings)
```

`loadExternalVariables` is defined in the same file and it returns a new data structure, so the `withExternalVariables` methods do not mutate state instead of copying.  There is no need for concern.